src/core/listener.c

```diff
@@ -767,6 +784,20 @@ QuicListenerParamSet(
             Listener->CibirId[1]);
 
         return QUIC_STATUS_SUCCESS;
+    } else if (Param == QUIC_PARAM_DOS_MODE_EVENTS) {
+        if (BufferLength > sizeof(BOOLEAN)) {
+            return QUIC_STATUS_INVALID_PARAMETER;
+        }
+
+        if (BufferLength == 0) {
+            //no value provided
+            return QUIC_STATUS_SUCCESS;
+        }
+
+        if (BufferLength == sizeof(BOOLEAN)) {
+            Listener->DosMitigationOptIn = *(BOOLEAN*)Buffer;
+        }
+        return QUIC_STATUS_SUCCESS;
```

Comment:
```suggestion
        if (BufferLength == sizeof(BOOLEAN)) {
            Listener->DosMitigationOptIn = *(BOOLEAN*)Buffer;
            return QUIC_STATUS_SUCCESS;
        }
```
Then it can just fallback to the bottom `return QUIC_STATUS_INVALID_PARAMETER;`

Reply:
Done